kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaCommandDecoder.java

```diff
@@ -168,18 +175,25 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
 
             CompletableFuture<AbstractResponse> responseFuture = new CompletableFuture<>();
             responseFuture.whenComplete((response, e) -> {
-                // Do nothing if it's triggered when responseFuture is expired
-                if (response == null) {
+                if (e != null) {
```

Comment:
we should check whether response is null, otherwise it will cause NPE in `responseToByteBuf` and `writeAndFlush` stage.

Reply:
OK

Reply:
I think it won't cause NPE, because the null `response` is never used. I'll just add the null check because the behavior is  unexpected. The response future should not be completed with null expect some error happened.